### PR TITLE
Install postcss with css to avoid warnings

### PR DESCRIPTION
**Pull request by tagliala**
### ⚠️ opinionated change

`css-loader` depends on `postcss`. `postcss-loader` instead has a peer
dependency on `postcss`.

This means that when following webpacker instructions to install
`postcss-loader` is installed, the console shows this warning:

```
warning " > postcss-loader@4.1.0" has unmet peer dependency "postcss@^7.0.0 || ^8.0.1".
```

On the other hand, installing `postcss` and `postcss-loader` together,
will result in another warning:

```
warning Pattern ["postcss@^8.2.1"] is trying to unpack in the same destination "node_modules/postcss" as pattern ["postcss@^8.1.4"]. This could result in non-deterministic behavior, skipping.
```

This change explicitly installs `postcss` with css packages to avoid
both warnings
How to replicate

```bash
$ rails new test-webpacker6-postcss --skip-active-record --skip-javascript
$ cd test-webpacker6-postcss/
$ echo "{ \"private\": true }" > package.json
$ echo "gem 'webpacker', git: 'https://github.com/rails/webpacker.git'" >> Gemfile
$ bundle
$ rails webpacker:install
$ yarn add css-loader mini-css-extract-plugin css-minimizer-webpack-plugin
$ yarn add postcss-loader
```

Output contains:
```
warning " > postcss-loader@4.1.0" has unmet peer dependency "postcss@^7.0.0 || ^8.0.1".
```

Ref:
- #2816 
- https://www.npmjs.com/package/css-loader?activeTab=dependencies
- https://www.npmjs.com/package/postcss-loader?activeTab=dependencies

### CODE REVIEW

**Comment by gauravtiwari** [comment]
thanks @tagliala 

but this should be alongside `yarn add postcss-loader postcss` right?

**Comment by tagliala** [comment]
> but this should be alongside yarn add postcss-loader postcss right?

Not in my case, I got the following warning because `postcss` was already there (installed by css-loader) 😢: 

```
warning Pattern ["postcss@^8.2.1"] is trying to unpack in the same destination "node_modules/postcss" as pattern ["postcss@^8.1.4"]. This could result in non-deterministic behavior, skipping.
```

Please try the following

```
$ rails new test-webpacker6-postcss --skip-active-record --skip-javascript
$ cd test-webpacker6-postcss/
$ echo "{}" > package.json
$ echo "gem 'webpacker', git: 'https://github.com/rails/webpacker.git'" >> Gemfile
$ rails webpacker:install
$ yarn add css-loader mini-css-extract-plugin css-minimizer-webpack-plugin
$ yarn add postcss-loader postcss
```

Apologies, I forgot to include my configuration. Maybe it is different with yarn v2

```
$ ruby -v
ruby 2.7.2p137 (2020-10-01 revision 5445e04352) [x86_64-darwin19

$ rails -v
Rails 6.1.0

$ node -v
v14.15.1

$ yarn -v
1.22.10
```

**Comment by javierav** [comment]
@tagliala I also have the same problem than you trying to update from webpacker 5...

**Comment by pedrofurtado** [comment]
Hey, @tagliala ! 👋 

That's true, the missing installation of `postcss` can be tricky depending of webpack configuration.

Can you please rebase/sync your PR to `master` branch?

After that, we will be ready to merge it ✅

**Comment by tagliala** [comment]
> Can you please rebase/sync your PR to `master` branch and move the code to this section (https://github.com/rails/webpacker/blob/090dbdbf326c0afabd6c4b08cc96e73caf510e3c/README.md#postcss)? To be together with `yarn add postcss-loader`.

Moving `postcss` together with `postcss-loader` will result in a different warning, as explained in the OP

**Review by lastthyme — APPROVED** [review]

**Comment by justin808** [comment]
This might change if we merge #3234. Let's wait on a decision for that first.

OTOH, it's great to have more documentation regarding the 6.0 upgrade!

**Comment by tagliala** [comment]
Thanks for the heads-up. When #3234 will be merged, I will take a look at the new documentation and check if this PR is still relevant

**Comment by justin808** [comment]
Feel free to reopen this PR here: 
https://github.com/shakacode/shakapacker/pulls

**Comment by tagliala** [comment]
Hi @justin808 , sure, will do that during the weekend so I can check if this issue is still relevant with the up-to-date documentation

**Pull request closed without merging**
